bin/compile

```diff
@@ -29,6 +29,16 @@ function indent() {
   esac
 }
 
+# Store which STACK we are running on in the cache to bust the cache if it changes
+if [ -f $CACHE_DIR/.apt/STACK ]; then
+  CACHED_STACK=$(cat "$CACHE_DIR/.apt/STACK")
+else
+  CACHED_STACK=$STACK
+  # Store the STACK in the cache for next time.
+  mkdir -p "$CACHE_DIR/.apt"
+  echo "$STACK" > "$CACHE_DIR/.apt/STACK"
```

Comment:
i think you want this to happen on every build (in case the stack has changed?). so maybe move it out of the `else`?